lib/elixir/lib/list.ex

```diff
@@ -410,9 +410,11 @@ defmodule List do
       []
 
   """
+  @spec wrap(improper_list) :: improper_list
+        when improper_list: nonempty_improper_list(any(), any())
```

Comment:
Here we are more precise than in the previous spec. Given an improper-list, it will return an improper list.